src/core/tsi/ssl_transport_security.c

```diff
@@ -1501,3 +1589,15 @@ int tsi_ssl_peer_matches_name(const tsi_peer *peer, const char *name) {
 
   return 0; /* Not found. */
 }
+
+/* --- Testing support. --- */
+const tsi_ssl_handshaker_factory_vtable *tsi_ssl_handshaker_factory_swap_vtable(
```

Comment:
hmm it seems that `tsi_ssl_handshaker_factory_swap_vtable()` is used only in `ssl_transport_security_test.c` file? then is it possible to declare/define it only in the `test.c` file instead of `ssl_transport_security.h` and `security_connector.c`? sorry if i missed something!

Reply:
The intention here is that normal users of `tsi_ssl_{client,server}handshaker_factory` will use `tsi_ssl_{client,server}_handshaker_factory_unref` to deference a handshaker factory. (The old way to was to call a `_destory` method directly.) Places like `security_connector.c` won't need to be concerned about the implementation details of the reference counting. Does that help?

Also, should make more sense with the forthcoming PR to update `security_connector.c`.

Reply:
hmm i'm not sure it understand you, e.g., how the unref/destroy functions are related to the swap_vtable function that is only used for testing... but i'll wait for the new PR :)